consensus/polybft/fsm.go

```diff
@@ -204,17 +204,33 @@ func (f *fsm) BuildProposal(currentRound uint64) ([]byte, error) {
 		return nil, err
 	}
 
-	if f.logger.IsDebug() {
+	if f.logger.GetLevel() <= hclog.Debug {
 		checkpointHash, err := extra.Checkpoint.Hash(f.backend.GetChainID(), f.Height(), stateBlock.Block.Hash())
 		if err != nil {
 			return nil, fmt.Errorf("failed to calculate proposal hash: %w", err)
 		}
 
-		f.logger.Debug("[FSM Build Proposal]",
-			"block", stateBlock.Block.Number(),
+		var buf bytes.Buffer
+
+		for i, tx := range stateBlock.Block.Transactions {
```

Comment:
Do this only if it is Debug or lower?

Reply:
Already done that: https://github.com/Ethernal-Tech/blade/blob/e250e5c21667e9ea700d29f80dfe945be84b0132/consensus/polybft/fsm.go#L207